odl/deform/linearized.py

```diff
@@ -0,0 +1,323 @@
+            ``displacement`` can be anything that is understood by the
+            ``domain.vector_field_space.element()`` method.
+        domain : `DiscreteLp`, optional
+            Space of templates on which this operator acts, i.e. the operator
+            domain. Default: ``displacement[0].space`` is used as domain.
+
+        Examples
+        --------
+        Create a given deformation and use it to deform a template.
+
+        Where the deformation field is zero we expect to get the same output
+        as the input. In the 4:th point, the deformation is non-zero and hence
+        we expect to get the value of the point 0.2 to the left, that is 1.0.
+
+        >>> import odl
+        >>> space = odl.uniform_discr(0, 1, 5)
+        >>> disp_field = space.vector_field_space.element([[0, 0, 0, -0.2, 0]])
+        >>> op = LinDeformFixedDisp(disp_field)
+        >>> template = [0, 0, 1, 0, 0]
+        >>> print(op([0, 0, 1, 0, 0]))
+        [0.0, 0.0, 1.0, 1.0, 0.0]
+
+        The result depends on the chosen interpolation. If we chose 'linear'
+        interpolation and offset the point half the distance between two
+        points, 0.1, we expect to get the mean of the values.
+
+        >>> space = odl.uniform_discr(0, 1, 5, interp='linear')
+        >>> disp_field = space.vector_field_space.element([[0, 0, 0, -0.1, 0]])
+        >>> op = LinDeformFixedDisp(disp_field)
+        >>> template = [0, 0, 1, 0, 0]
+        >>> print(op(template))
+        [0.0, 0.0, 1.0, 0.5, 0.0]
+        """
+        if domain is None:
+            if not isinstance(displacement.space, ProductSpace):
+                raise TypeError('`displacement.space` {!r} not a '
+                                '`ProductSpace`'.format(displacement.space))
+            if not displacement.space.is_power_space:
+                raise TypeError('`displacement.space` {!r} not a power'
+                                'space'.format(displacement.space))
+            if not isinstance(displacement[0].space, DiscreteLp):
+                raise TypeError('`displacement[0].space` {!r} not an '
+                                '`DiscreteLp`'.format(displacement[0]))
+
+            domain = displacement[0].space
+        else:
+            if not isinstance(domain, DiscreteLp):
+                raise TypeError('`displacement[0]` {!r} not an `DiscreteLp`'
+                                ''.format(displacement[0]))
+
+            displacement = domain.vector_field_space.element(displacement)
+
+        Operator.__init__(self, domain, domain, linear=True)
+
+        self._displacement = displacement
+
+    def _call(self, template, out=None):
+        """Implementation of ``self(template)``.
+
+        Parameters
+        ----------
+        template : `domain` element
+            Given template that is to be deformed by the fixed
+            displacement field.
+        """
+        return _linear_deform(template, self._displacement, out)
+
```

Comment:
Short comment here, don't we have a formula for the inverse operator as well? Perhaps it should be added.